lib/emoji.rb

```diff
@@ -77,15 +77,23 @@ def parse_data_file
         JSON.parse(file.read, symbolize_names: true)
       end
 
+      if "".respond_to?(:-@)
+        # Ruby >= 2.3 this is equivalent to .freeze
+        # Ruby >= 2.5 this will freeze and dedup
+        dedup = lambda { |str| -str }
+      else
+        dedup = lambda { |str| str.freeze }
+      end
+
       append_unicode = lambda do |emoji, raw|
         unless TEXT_GLYPHS.include?(raw) || emoji.unicode_aliases.include?(raw)
-          emoji.add_unicode_alias(raw)
+          emoji.add_unicode_alias(dedup.call(raw))
```

Comment:
Each unicode alias is supposed to be unique, i.e. two emoji cannot share the same unicode alias. Does this line really benefit from deduping?

Reply:
Removing this dedup adds about 2000 live objects 🤔 Looking into why...

Reply:
Ah, this makes sense. We're using the unicode aliases as keys in a hash, for which Ruby will automatically duplicate and freeze those strings.

```
> hash = {}
> str = "foo"
=> "foo"
> hash[str] = "bar"
=> "bar"
> hash.keys[0]
=> "foo"
> hash.keys[0].frozen?
=> true
> str.frozen?
=> false
> hash.keys[0].object_id
=> 70288572681220
> str.object_id
=> 70288572680560
```

So we were already implicitly deduping these for the hash, but were keeping the unfrozen/undeduped copy around in the unicode_aliases array.